model/project.go

```diff
@@ -1302,8 +1302,8 @@ func FindLatestVersionWithValidProject(projectId string) (*Version, *Project, *P
 				revisionOrderNum = lastGoodVersion.RevisionOrderNumber // look for an older version if the returned version is malformed
 				continue
 			}
+			return lastGoodVersion, project, pp, nil
```

Comment:
This was an unrelated change I found when my test fails -- we shouldn't be returning the first iteration of the loop, we should return the error outside the loop. Wrote a new test to catch this.